internal/runtime/client/client_test.go

```diff
@@ -380,257 +372,336 @@ func TestClient_CallExtension(t *testing.T) {
 		},
 	}
 
-	registryWithExtensionHandlerWithIgnorePolicy := runtimeregistry.New()
-	err = registryWithExtensionHandlerWithIgnorePolicy.WarmUp(&runtimev1.ExtensionConfigList{
-		Items: []runtimev1.ExtensionConfig{
-			validExtensionHandlerWithIgnorePolicy,
-		},
-	})
-	g.Expect(err).NotTo(HaveOccurred())
-
-	successResponse := &fakev1alpha1.FakeResponse{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "FakeResponse",
-			APIVersion: fakev1alpha1.GroupVersion.String(),
-		},
-		CommonResponse: runtimehooksv1.CommonResponse{
-			Status: runtimehooksv1.ResponseStatusSuccess,
-		},
-	}
-
-	failureResponse := &fakev1alpha1.FakeResponse{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "FakeResponse",
-			APIVersion: fakev1alpha1.GroupVersion.String(),
-		},
-		CommonResponse: runtimehooksv1.CommonResponse{
-			Status: runtimehooksv1.ResponseStatusFailure,
-		},
-	}
-
 	type args struct {
 		hook     runtimecatalog.Hook
 		name     string
 		request  runtime.Object
 		response runtimehooksv1.ResponseObject
 	}
 	tests := []struct {
-		name       string
-		registry   runtimeregistry.ExtensionRegistry
-		args       args
-		testServer testServerConfig
-		wantErr    bool
+		name                       string
```

Comment:
How about adding a test to check the call fails if the registered GH and the passed GH do not match? This is a new check we added recently in the `CallExtension` code.

Reply:
Good point, done